.NET Standard TOC: combined parallel programming topics

pkulikov opened the pull request:
This PR is based on two observations:
- [.NET Guide](https://docs.microsoft.com/en-us/dotnet/standard/) TOC is long and has four entries related to parallel programming.
- [One of those entries](https://docs.microsoft.com/en-us/dotnet/standard/parallel-processing-and-concurrency) is an entry point to other three (not totally correct here, though).

I thought: "Oh, there is an opportunity to compact TOC!" Was it a good thought, or having various parallel-code related topics on one level was intentional?

Additionally
- Replaced *.NET Framework* with *.NET*
- Recommended TAP instead of EAP in the [Threading](https://docs.microsoft.com/en-us/dotnet/standard/threading/) topic.

pkulikov committed:
.NET standard TOC: combined parallel programming topics

rpetrusha reviewed (changes requested):
Thank you so much for submitting this PR, @pkulikov. You've really done an outstanding job of adding clarity, particularly to the top-level async topic. What I'm a bit concerned about is the TOC -- do developers ordinarily associate parallel processing and concurrency with async? So perhaps the title of the top-level Parallel Processing and Concurrency in .NET topic should be renamed to "Parallel Processing, Concurrency, and Async Programming in .NET" for the title, and "Parallel Processing, Concurrency, and Async" for the TOC. Let me know what you think about that -- I'm fine either with changing it or leaving it as it is in your PR.

pkulikov committed:
Feedback

pkulikov commented:
@rpetrusha thank you for the remark. I've had the same concern. Please check my latest commit with the fix for that. I've also added description and updated introduction text, so please review that as well.

rpetrusha reviewed (approved):
Thanks for making the additional changes, @pkulikov. This looks extremely good! I'll merge it now. It should be live in the next day or two.
